.github/workflows/main.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build and Deploy
+on: [push]
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./pages
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
+        with:
+          persist-credentials: false
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/.next/cache
+          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Install and Build 🔧 # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
+        uses: actions/setup-node@v1
+      - run: npm install
+      - run: npm run export
+        env:
+            CI: true
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.6.2
+        with:
+          GITHUB_TOKEN: ${{ secrets.DOCS_DEPLOY_TOKEN }}
+          REPOSITORY_NAME: braden-wilson/braden-wilson.github.io
+          BRANCH: master # The branch the action should deploy to.
+          FOLDER: out # The folder the action should deploy.
```

Comment:
what does this do? Does it push the output to repo `braden-wilson/braden-wilson.github.io`?

Reply:
Yeah, every time there's a push it builds the website in `braden-wilson/braden-wilson.github.io` because that's where I'm testing github pages from. Should this be removed or changed for the PR?

Reply:
it should be on pull request, not during push

Reply:
actually I am wrong oops, it should be on push for only the `main` branch